### PR TITLE
fix: rescale loot tracker kc for nothing drop rate

### PR DESCRIPTION
**Pull request by iProdigy**
Follow-up to https://github.com/pajlads/DinkPlugin/pull/427#issuecomment-1974982089

Caused by https://github.com/runelite/runelite/issues/5077 / https://github.com/runelite/runelite/issues/10831 (dupe)

&nbsp;

<sup>Also fixes a bug where drops might not be found for an NPC that has some parenthetical suffix appended by the wiki such as https://oldschool.runescape.wiki/w/TzHaar-Mej_(monster) - this only applies for a handful of monsters but achieves 87kB (9kB compressed) reduction in npc_drops.json due to removing monster variations that would never be matched</sup>

### CODE REVIEW

**Comment by iProdigy** [comment]
Works* when tested in-game

loot tracker reports 11kc

dink reports 44kc

*Flesh crawlers are a special case where the 'Nothing' drop rate is overestimated due to the [wiki](https://oldschool.runescape.wiki/w/Flesh_Crawler#Other) reporting separate rates for f2p/p2p - this typically isn't an issue for other monsters with Nothing as a possibility

File: src/test/java/dinkplugin/RarityCalculator.java [hunk 1]
```diff
@@ -140,7 +150,8 @@ public Transformed transform() {
                     denom = BigDecimal.valueOf(2000);
                     break;
                 default:
-                    String[] parts = StringUtils.split(cleanRarity, '/');
+                    String fraction = cleanRarity.endsWith("%") ? cleanRarity.substring(0, cleanRarity.length() - 1) + "/100" : cleanRarity;
```
Comment (iProdigy) [inline src/test/java/dinkplugin/RarityCalculator.java:153]: won't be necessary in next wiki dump; fixed in https://oldschool.runescape.wiki/?title=TzHaar-Mej_%28monster%29&diff=14552108&oldid=14523815 (unless someone makes the same mistake elsewhere)

**Review by pajlada — APPROVED** [review]
Not tested but LGTM :+1:

**Review by Felanbird — APPROVED** [review]
probably fine, flesh crawlers were unhinged as expected 11->48, probably closer to 15 actually killed

i only did aberrant spectres to 52 KC but it didn't add any, maybe not enough kc to factor in their nothing drop

**Comment by iProdigy** [comment]
> i only did aberrant spectres to 52 KC but it didn't add any, maybe not enough kc to factor in their nothing drop

oops, abby spec has an Always drop in certain circumstances (but the data source assumes the circumstance applies), so this wipes out the Nothing drop (which results in no kc adjustment)

**Pull request merged**